i3_agenda/API.py

```diff
@@ -0,0 +1,131 @@
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(credspath, SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open(TMP_TOKEN, "wb") as token:
+            pickle.dump(creds, token)
+
+    service = build("calendar", "v3", credentials=creds)
+    return service
+
+
+def getEvents(
+        credentials: str, allowed_calendars_ids: List[str], max_results: int, today_only=False
+) -> List[Event]:
+    service = connect(credentials)
+
+    now = datetime.datetime.utcnow()
+    now_rfc3339 = now.isoformat() + "Z"  # 'Z' indicates UTC time
+    calendar_ids = []
+    while True:
+        calendar_list = service.calendarList().list().execute()
+        for calendar_list_entry in calendar_list["items"]:
+            if (
+                    not allowed_calendars_ids
+                    or calendar_list_entry["id"] in allowed_calendars_ids
+            ):
+                calendar_ids.append(calendar_list_entry["id"])
+        page_token = calendar_list.get("nextPageToken")
+        if not page_token:
+            break
+
+    all_events = []
+    for calendar_id in calendar_ids:
+        if today_only:
+            midnight_rfc3339 = (
+                    now.replace(hour=23, minute=39, second=59).isoformat() + "Z"
+            )
+            events_result = (
+                service.events()
+                .list(
+                    calendarId=calendar_id,
+                    timeMin=now_rfc3339,
+                    timeMax=midnight_rfc3339,
+                    maxResults=max_results,
+                    singleEvents=True,
+                    orderBy="startTime",
+                )
+                .execute()
+            )
+        else:
+            events_result = (
+                service.events()
+                .list(
+                    calendarId=calendar_id,
+                    timeMin=now_rfc3339,
+                    maxResults=max_results,
+                    singleEvents=True,
+                    orderBy="startTime",
+                )
+                .execute()
+            )
+        events = events_result.get("items", [])
+
+        if not events:
+            continue
+
+        for event in events:
+            end_time = get_event_time(
+                event["end"].get("dateTime", event["end"].get("date"))
+            )
+            start_time = event["start"].get("dateTime", event["start"].get("date"))
+            unix_time = get_event_time(start_time)
+
+            location = None
+
+            if "location" in event:
+                location = event["location"]
+            elif "description" in event:
+                matches = re.findall(URL_REGEX, event["description"])
+                location = matches[0][0] if matches else None
+
+            all_events.append(
+                Event(
+                    event["summary"],
+                    is_allday(start_time),
```

Comment:
instead of passing this here, the `Event` class could have a `is_allday` method and infer the value of `is_allday` in the constructor  (the __init__ method)

Reply:
Using a unix time to infer that would be something like if the time of the event is 00:00:00 so it is all day. Maybe if there is a event that begins at midnight but isn't all day it will infer it wrong. Despite that, I guess that is a good approach for now. What do you think?

Reply:
I see what you mean, because we don't pass the `start_time` to the constructor.  
The way of checking with 00:00:00 is incorrect because events can start at 00:00:00 and not be all day.  
I see 2 options:
1. pass the `start_time` to the constructor and add it to the cache. The problem with it might be that users that currently use this program will have "corrupted" cache files, we could just delete them if they are invalid when trying to read them, so no big deal.
2. Make the `is_allday` a static method of Event and essentially make it work like before, just move it